distributed-system-project/http-blob-server/blob/services/blob.service.ts

```diff
@@ -0,0 +1,107 @@
+import { mkdir, readFile, rm, writeFile } from "node:fs/promises";
+import { createReadStream, existsSync } from "node:fs";
+import { config } from "../../config.ts";
+import { getFullFileDir } from "../utils/filesystem.ts";
+import { extractRawContent } from "../helpers.ts";
+import type { BlobMetadata } from "../types.ts";
+import { logger } from "../../../logger/index.ts";
+
+export class GetBlobError extends Error {
+  constructor(message?: string) {
```

Comment:
since node 24, we can add `options.cause` to the error's constructor. the project explicitly states v23+, so just fyi.
this applies to all custom errors in this project.